### PR TITLE
Allow configuration of s3 as filestore.backend

### PR DESCRIPTION
**Pull request by janraasch**
Follow-up to https://github.com/getsentry/onpremise/pull/74.

@mattrobenolt, I am not sure, if `8.22` is the right place for these changes. Let me know, if there is anything that needs to be done so this can get merged.

### CODE REVIEW

**Comment by regner** [comment]
Why only S3? The docs for sentry say gcs is supported as well: https://docs.sentry.io/server/filestore/

**Comment by mattrobenolt** [comment]
gcs was added like, last week. We don't have any released versions with gcs support yet. That'll come with 9.1.

**Comment by regner** [comment]
hahaha OK, thanks. Just looking at Sentry now and didn't realize that was a new feature. :)

File: 8.22/sentry.conf.py [hunk 1]
```diff
@@ -223,10 +228,24 @@
...
+    SENTRY_OPTIONS['filestore.backend'] = 's3'
+    SENTRY_OPTIONS['filestore.options'] = {
+        'access_key': env('SENTRY_FILESTORE_ACCESS_KEY'),
+        'bucket_name': env('SENTRY_FILESTORE_BUCKET_NAME'),
+        'secret_key': env('SENTRY_FILESTORE_SECRET_KEY'),
+        # https://github.com/getsentry/sentry/issues/6598
+        'signature_version': env('SENTRY_FILESTORE_SIGNATURE')
+    }
+else:
+  raise Exception('Error: SENTRY_FILESTORE_BACKEND only supports `S3` or `FILESYSTEM`')
```
Comment (Tenzer) [inline 8.22/sentry.conf.py]: This line should be indented with 4 spaces in order to match the rest of the file.

Comment (janraasch) [reply 8.22/sentry.conf.py]: Absolutely 🙌. I ammended my commit with a fix and rebased on current master. Please recheck.

Comment (Tenzer) [reply 8.22/sentry.conf.py]: Looks good to me now.

**Comment by ajacques** [comment]
+1 for this change.

Would it make more sense to add this to 9.0 since 8.22 is not the latest version or add it to both if you're still on 8.22?

Additionally, are we able to use the EC2 instance profile instead of static credentials with this PR? It looks like Sentry internally supports this [here](https://github.com/getsentry/sentry/blob/master/src/sentry/filestore/s3.py#L336-L347), so we should be able to use this by not specifying any parameter for the env variable `SENTRY_FILESTORE_ACCESS_KEY`, right?

**Comment by janraasch** [comment]
For our use case, it would be neat to have a `8.x` version with this change, but if not, I would also not mind having to upgrade. We use a fork at the moment any way...

**Comment by janraasch** [comment]
@Tenzer Is there anything else I could help with to get this merged? I would love to get this dealt with. It's been almost a year 😇.

**Comment by Tenzer** [comment]
I'm just a user of Sentry and don't have permissions to merge stuff on this repository. Sorry.

**Comment by Tenzer** [comment]
Looking at the most recently merged PRs, it's @mattrobenolt who's been doing that.

**Comment by janraasch** [comment]
Thank you @Tenzer for the quick reply. Dear @mattrobenolt 😇, hi there, nice to meet you. I was wondering, if there is anything else I could help with to get this merged? This pr has been open for quite a while *and* it was approved by 2 people 😊.

**Comment by BYK** [comment]
Hi @janraasch! Thanks a lot for the PR but we are trying to limit the env variables we actively support to a minimum core. For anything more than that you can still modify the config file yourself, which is what we encourage. This is easier if you are using the [getsentry/onpremise](https://github.com/getsentry/onpremise) repo but still not too hard if you want to use this image and roll your own setup (I'd say it makes even more sense in that case as you are going barebones).

For this reason and due to the staleness of this PR, I'm closing it.

**Pull request closed without merging**